### ISSUE TITLE
karbon plugin completely over writes ~/.kube/config by default

### ISSUE THREAD

**Issue by nelaaro** [nutanix/kubectl-karbon#12]
As a default overwriting ~/.kube/config is not a safe way to do things.
I lost all my saved config and clusters that I was running out of there. 

At the very least warn people and give them a chance to backup the configs before completing the login. 

```
kubectl karbon login
this is destructive and will overwrite your ~/.kube/config
```

Example of what happens on the default config and installation. 

```
kubectl config get-contexts
CURRENT   NAME                                                   CLUSTER                                                AUTHINFO                                               NAMESPACE
          arn:aws:eks:ClustA                            arn:aws:eks:CLustA                                          arn:aws:eks:ClustA
*         clustB-prod-context                                   clustB-prod                                default-user-clustB-prod                   registry
          clustC-prod                                        clustC-prod                                        kube-admin-clustC-prod
          clustD-prod                                        clustD-prod                                        kube-admin-clustD-prod

kubectl karbon login
Enter admin@domain.com password:
Logged successfully into clustB-prod cluster

kubectl get nodes
NAME                                                 STATUS   ROLES    AGE   VERSION
karbon-clustB-prod-2d7780-k8s-master-0   Ready    master   8d    v1.20.9
karbon-clustB-prod-2d7780-k8s-master-1   Ready    master   8d    v1.20.9
karbon-clustB-prod-2d7780-k8s-worker-0   Ready    node     8d    v1.20.9
karbon-clustB-prod-2d7780-k8s-worker-1   Ready    node     8d    v1.20.9
karbon-clustB-prod-2d7780-k8s-worker-2   Ready    node     8d    v1.20.9
karbon-clustB-prod-2d7780-k8s-worker-3   Ready    node     8d    v1.20.9
karbon-clustB-prod-2d7780-k8s-worker-4   Ready    node     8d    v1.20.9

kubectl config get-contexts
CURRENT   NAME                              CLUSTER                   AUTHINFO                               NAMESPACE
*         clustB-prod-context   clustB-prod   default-user-clustB-prod
````

I get you have a Kubie tool integration whatever that means. 
You shouldn't overwrite the whole config file to update the user token and cluster details.
Or you should put the config file into a location that is not part of the standard `~/.kube/config` that is the default used
by other applications. 
kubectl operation and functions should not change because of how you handle the ~/.kube/config file. 
Update the cluster-specific details in the `kube/config` file or leave it alone. never overwrite the whole file. 

The defaults in the `~/.kubectl-karbon.yaml` file should prevent the `~/.kube/config` file being changed in an irreversible way. 
```
 cat ~/.kubectl-karbon.yaml
# karbon config
# https://github.com/nutanix/kubectl-karbon

kubie: true
kubeconfig: ~/.kube/karbon/config
# or
kubeconfig: ~/.kube/kubie/config.karbon.yml
```

This makes more sense in place of overwriting the defaults

**Comment by tuxtof** [comment]
Hello @nelaaro , sorry for your misfortune
but we try to say it correctly in the readme

kubectl karbon login Authenticate user with Nutanix Prism Central, **create** kubeconfig file, get ssh key/cert

after this is your choice to specify where to create it and by default we use the default kuebconfig path

but it seems that it is not clear enough, in all case i hope if your file contain important data you were able to restore it from a backup ???

people use all kubeconfig files in different ways 

kubie mode for exemple store each cluster kubeconfig in a different file in a common folder (default is .kubectl/kubie/ but it can be change with kubie-path var) the idea was to combine it with [Kubie tool](https://blog.sbstp.ca/introducing-kubie/), this is also explained in the Readme

in all case the initial goal is to provide an out of the box experience where user can directly use their cluster after the karbon login and for advanced user like you to have the settings to adapt the behavior
but your are right i will think to implement an overwrite warning in a futur version

**Comment by tuxtof** [comment]
fixed with #14 in release v0.7.0
